File: Hashing/php/allUnique.php
<?php

declare(strict_types=1);

function allUnique(array $items): bool
{
    $seen = [];
    foreach ($items as $item) {
        $key = serialize($item);
        if (isset($seen[$key])) {
            return false;
        }
        $seen[$key] = true;
    }

    return true;
}

File: Hashing/csharp/AllUnique.cs
using System.Collections.Generic;

namespace HashingSolutions;

public static class AllUnique
{
    public static bool Solve<T>(IEnumerable<T> items)
    {
        var seen = new HashSet<T>();
        foreach (var item in items)
        {
            if (!seen.Add(item))
            {
                return false;
            }
        }

        return true;
    }
}
